packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -1521,26 +1525,17 @@ export async function saveFile(doc: vscode.TextDocument, datasetProvider: api.IZ
     const ending = doc.fileName.substring(start);
     const sesName = ending.substring(0, ending.indexOf(path.sep));
     const profile = Profiles.getInstance().loadNamedProfile(sesName);
-    const fileLabel = path.basename(doc.fileName);
-    const dataSetName = fileLabel.substring(0, fileLabel.indexOf("("));
-    const memberName = fileLabel.substring(fileLabel.indexOf("(") + 1, fileLabel.indexOf(")"));
     if (!profile) {
         const sessionError = localize("saveFile.session.error", "Could not locate session when saving data set.");
         ZoweLogger.error(sessionError);
         await api.Gui.errorMessage(sessionError);
         return;
     }
 
-    const etagFavorites = (
-        datasetProvider.mFavorites
-            .find((child) => child.label.toString().trim() === sesName)
-            ?.children.find((child) => child.label.toString().trim() === dataSetName)
-            ?.children.find((child) => child.label.toString().trim() === memberName) as api.IZoweDatasetTreeNode
-    )?.getEtag();
+    // TODO Handle case where same data set is open as both favorite and non-favorite to prevent desync between equivalent nodes
```

Comment:
Having the same data set open both as a favorite and non-favorite is probably enough of an edge case that we can wait to address it in v3. Discussed with @traeok and we believe the FileSystemProvider should fix the desync since it stores data per URI rather than per node 😋